cwrc_preserver.rb

```diff
@@ -47,67 +79,86 @@ module CWRCPerserver
   raise CWRCArchivingError if swift_depositer.nil?
 
   # get list of all objects from cwrc
-  cwrc_objs = if re_process
+  # returns an array of objects (e.g., [{"pid"=>"1"}, {"pid"=>"2"}])
+  cwrc_objs = if !reprocess.empty?
                 rp_list = []
-                File.open(ENV['CWRC_REPROCESS']).each { |line| rp_list << line } if File.exist?(ENV['CWRC_REPROCESS'])
-                Hash(rp_list.collect { |v| ['pid', v] })
+                log.debug("Processing ID list from file: #{reprocess}")
+                File.open(reprocess).each { |line| rp_list << { 'pid' => line.strip } } if File.exist?(reprocess)
               else
+                log.debug('Processing api response')
                 get_cwrc_objs(cookie, start_dt)
               end
-  log.debug("Number of objects to precess: #{cwrc_objs&.length}")
+  log.debug("Number of objects to process: #{cwrc_objs.nil? ? '0' : cwrc_objs&.length}")
 
   # for each cwrc object
   cwrc_objs&.each do |cwrc_obj|
-    cwrc_file_str = cwrc_obj['pid'].to_s
-    cwrc_file = "#{cwrc_file_str.tr(':', '_')}.zip"
+    cwrc_file = "#{cwrc_obj['pid'].tr(':', '_')}.zip"
+    cwrc_file_tmp_path = File.join(work_dir, cwrc_file)
 
-    log.debug("PROCESSING OBJECT: #{cwrc_file_str}, modified timestamp #{cwrc_obj['timestamp']}")
+    log.debug("PROCESSING OBJECT: #{cwrc_obj['pid']}, modified timestamp #{cwrc_obj['timestamp']}")
 
     # check if file has been deposited, handle open stack bug causing exception in openstack/connection
-    force_deposit = false || re_process
+    force_deposit = false || !reprocess.empty?
     begin
       swift_file = swift_depositer.get_file_from_swit(cwrc_file, ENV['CWRC_SWIFT_CONTAINER']) unless force_deposit
-    rescue StandardError
+    rescue StandardError => e
       force_deposit = true
+      log.debug("Force deposit in swift: #{cwrc_obj['pid']} #{e.message}")
     end
 
-    # if object is not is swift or we have newer object
-    next unless force_deposit || swift_file.nil? || swift_file.metadata['timestamp'].nil? ||
+    # if object does not exist within Swift, or is outdated, or has a zero size, or has been marked for a forced update
+    next unless force_deposit ||
+                swift_file.nil? ||
+                swift_file.bytes.to_f.zero? ||
+                swift_file.metadata['timestamp'].nil? ||
                 cwrc_obj['timestamp'].to_s.to_time > swift_file.metadata['timestamp'].to_s.to_time
```

Comment:
`to_s.to_time`: the `to_s` is redundant, correct? The original code was littered with `.to_s` on Strings  so I'm double-checking.

Reply:
I think the cwrc_obj['timestamp'] is already a string? this is to convert the string to a time object. to_s is redundant.